crates/sdk/src/wallet/derivation_path.rs

```diff
@@ -44,9 +51,102 @@ impl DerivationPath {
         }
     }
 
+    pub fn has_shielded_compatible_coin_type(&self) -> bool {
+        if let Some(coin_type) = self.0.as_ref().get(1) {
+            coin_type.to_u32() == NAMADA_COIN_TYPE
+        } else {
+            true
+        }
+    }
+
+    /// Check if the path is BIP-0044 conform
+    /// https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#path-levels
+    pub fn is_bip44_conform(&self, strict: bool) -> bool {
+        // check the path conforms the structure:
+        // m / purpose' / coin_type' / account' / change / address_index
+        let purpose = self.0.as_ref().get(0);
+        let coin_type = self.0.as_ref().get(1);
+        let account = self.0.as_ref().get(2);
+        let change = self.0.as_ref().get(3);
+        let address = self.0.as_ref().get(4);
+        let junk = self.0.as_ref().get(5);
+        if let (
+            Some(purpose),
+            Some(coin_type),
+            Some(account),
+            Some(change),
+            Some(address),
+            None,
+        ) = (purpose, coin_type, account, change, address, junk)
+        {
+            purpose.to_u32() == BIP44_PURPOSE
+                && purpose.is_hardened()
+                && coin_type.is_hardened()
+                && account.is_hardened()
+                && (!strict || (change.is_normal() && address.is_normal()))
+        } else {
+            false
+        }
+    }
+
+    /// Check if the path is SLIP-0010 conform
+    /// https://github.com/satoshilabs/slips/blob/master/slip-0010.md#child-key-derivation-ckd-functions
+    pub fn is_slip10_conform(&self, scheme: SchemeType) -> bool {
+        match scheme {
+            SchemeType::Ed25519 => {
+                // all indices must be hardened
+                self.0.as_ref().iter().all(|idx| idx.is_hardened())
+            }
+            _ => true,
```

Comment:
is secp256k1 always conforming?

Reply:
There is no constraint indeed for secp256k1, according to https://github.com/satoshilabs/slips/blob/master/slip-0010.md#child-key-derivation-ckd-functions
I'll add a comment for this line. Thanks!